src/DDTrace/Integrations/Predis/PredisSandboxedIntegration.php

```diff
@@ -0,0 +1,217 @@
+            $span->service = 'redis';
+            $span->resource = 'Predis.Client.connect';
+            PredisSandboxedIntegration::setConnectionTags($this, $span);
+        });
+
+        \dd_trace_method('Predis\Client', 'executeCommand', function (SpanData $span, $args) use ($integration) {
+            $span->name = 'Predis.Client.executeCommand';
+            $span->type = Type::CACHE;
+            $span->service = 'redis';
+            PredisSandboxedIntegration::setConnectionTags($this, $span);
+            $integration->addTraceAnalyticsIfEnabled($span);
+
+            // We default resource name to 'Predis.Client.executeCommand', but if we are able below to extract the query
+            // then we replace it with the query
+            $span->resource = 'Predis.Client.executeCommand';
+
+            if (\count($args) == 0) {
+                return;
+            }
+
+            $command = $args[0];
+            $arguments = $command->getArguments();
+            array_unshift($arguments, $command->getId());
+            $span->meta['redis.args_length'] = count($arguments);
+            $query = PredisSandboxedIntegration::formatArguments($arguments);
+            $span->resource = $query;
+            $span->meta['redis.raw_command'] = $query;
+        });
+
+        \dd_trace_method('Predis\Client', 'executeRaw', function (SpanData $span, $args) use ($integration) {
+            $span->name = 'Predis.Client.executeRaw';
+            $span->type = Type::CACHE;
+            $span->service = 'redis';
+            PredisSandboxedIntegration::setConnectionTags($this, $span);
+            $integration->addTraceAnalyticsIfEnabled($span);
+
+            // We default resource name to 'Predis.Client.executeRaw', but if we are able below to extract the query
+            // then we replace it with the query
+            $span->resource = 'Predis.Client.executeRaw';
+
+            if (\count($args) == 0) {
+                return;
+            }
+            $arguments = $args[0];
+            $query = PredisSandboxedIntegration::formatArguments($arguments);
+            $span->resource = $query;
+            $span->meta['redis.args_length'] = count($arguments);
+            $span->meta['redis.raw_command'] = $query;
+        });
+
+        \dd_trace_method('Predis\Pipeline\Pipeline', 'executePipeline', [
+
+            'prehook' => function (SpanData $span, $args) {
+                $span->name = 'Predis.Pipeline.executePipeline';
+                $span->resource = $span->name;
+                $span->type = Type::CACHE;
+                $span->service = 'redis';
+                PredisSandboxedIntegration::setConnectionTags($this, $span);
+                if (\count($args) < 2) {
+                    return;
+                }
+                $commands = $args[1];
+                $span->meta['redis.pipeline_length'] = count($commands);
+            },
+        ]);
+
+        return SandboxedIntegration::LOADED;
+    }
+
+    /**
```

Comment:
I decided to duplicate the functions here instead of extracting as:

- I propose to remove the non-sandboxed integrations very soon
- only 2 out of 3 are exactly the same
- the overhead of loading yet one class when we know that the non sandboxed integration will go soon
- we still have tests (they are shared) to make sure that they do not go out of sync